### PR TITLE
Create TaskManager

### PR DESCRIPTION
**Pull request by Ewan10**

### CODE REVIEW

**Comment by thompsy** [comment]
It would be good practice to run some basic formatting and linting on your code. Here's a bit of an intro to how you could do some of this with VS Code: https://code.visualstudio.com/docs/java/java-linting

**Comment by thompsy** [comment]
It would also be worth updating the name of the file to include the `.java` extension. This allows tools like Github to recognise what language it is and do syntax highlighting automatically.

**Comment by thompsy** [comment]
You might want to look at using Maven to build the project. This should be a good place to start: http://maven.apache.org/guides/getting-started/maven-in-five-minutes.html

**Review by thompsy — CHANGES_REQUESTED** [review]

**Comment by thompsy** [comment]
One more note for now: it would be worth updating your comments. The long dashes aren't really necessary. This page has some sensible guidance for good coding practices generally and for commenting specifically: https://www.baeldung.com/java-clean-code

File: TaskManager.java [hunk 1]
```diff
@@ -0,0 +1,217 @@
...
+
+class TaskManager {
+    private String title;
+    private String date;
+    private List<Task> LT= new ArrayList<Task>();
+
+//____________________Helper function DateComparison() This method compares the date of a stored 
+//object with the date of today from the Date() object. It is used by the DueToday() method.
+
+	public boolean DateComparison(String date) {
```
[the remark this answers is not in the document]
Comment (Ewan10) [reply TaskManager.java:36]: I will revise the date and time classes. Of course.

File: TaskManager.java [hunk 2]
```diff
@@ -0,0 +1,217 @@
+import java.util.*;
+import java.util.regex.*;
+
+class Task {
+    private String title;
+    private String date;
```
[the remark this answers is not in the document]
Comment (Ewan10) [reply TaskManager.java:6]: I got to do revision on the date/time classes for java and amend those parts.

File: TaskManager.java [hunk 3]
```diff
@@ -0,0 +1,217 @@
...
+	  	  
+       	     while(!(userDate[0] <=31 && userDate[0] >= 1) || (!(userDate[1] <=12 && userDate[1] >= 1)) 
+       	     || (!(userDate[2] >= 2021)));
+         
+             LT.add(new Task(title, date));
+    } 
+
+//______________The ChangeDate() method used to change the date of an already stored task.
+
+    public void  ChangeDate() {
```
[the remark this answers is not in the document]
Comment (Ewan10) [reply TaskManager.java:136]: I 'll fix that as well. I agree to conform with the convention.

File: TaskManager.java [hunk 4]
```diff
@@ -0,0 +1,217 @@
...
+    public static void main(String[] args) {
+      	  TaskManager T1 = new TaskManager();
+          System.out.println("For creating a new task type add \n");
+          System.out.println("For viewing the tasks due today type today \n");
+          System.out.println("For viewing all tasks type all \n");
+          System.out.println("For changing the date of a task type change date \n");
+          System.out.println("For marking a task as complete type \"complete\"\n"); 
+          Scanner sc = new Scanner(System.in);
+          
+          while(true) {
```
[the remark this answers is not in the document]
Comment (Ewan10) [reply TaskManager.java:198]: I already studied the command line parsing from apache commons. I am aware of the definition stage, the parsing stage and the interrogations stage by creating an Options object, DefaultParser, adding options and checking the user input.

[the remark this answers is not in the document]
Comment (Ewan10) [reply TaskManager.java:198]: I have noted down all the points for improvement. The use of the date/time class, the separation of concerns for two methods which I noticed during the development and debugging phase, the apppropriate naming also for various variables and methods, the command line parsing and the permanent storage of the tasks. I 'll work on each of them step by step and I 'll upload a new version of the project.

File: TaskManager.java [hunk 5]
```diff
@@ -0,0 +1,217 @@
...
+//___________________The Complete() method for marking a task as complete.
+
+     public void Complete() {
+        Scanner sx = new Scanner(System.in);
+        System.out.println("Enter a task title: ");
+        title = sx.nextLine();    
+       
+        for(Task ta: LT) {
+         if(title.equals(ta.getTitle())) {
+            ta.setTitle(" -Done.");
```
[the remark this answers is not in the document]
Comment (Ewan10) [reply TaskManager.java:180]: I can create a new field. I will change it.

File: TaskManager.java [hunk 6]
```diff
@@ -0,0 +1,217 @@
...
+                t.print();
+            }
+        }
+        if(b== false) {
+            System.out.println("This task does not exist."); }
+    }
+
+//______________________________The All() method for viewing all the stored tasks.
+
+    public void All() {
```
[the remark this answers is not in the document]
Comment (Ewan10) [reply TaskManager.java:157]: I 'll just rename it viewAllTasks().

File: TaskManager.java [hunk 7]
```diff
@@ -0,0 +1,217 @@
...
+    } 
+    public void setDate(String s) {
+        date = s;
+    }
+    public void print() { System.out.println("title: " + title + " date: " + date); }
+}
+//_________________TaskManager class_____________________________________
+
+class TaskManager {
+    private String title;
```
[the remark this answers is not in the document]
Comment (Ewan10) [reply TaskManager.java:29]: I 'll just remove them.

File: TaskManager.java (hunk 4 above)
[the remark this answers is not in the document]
Comment (thompsy) [reply TaskManager.java:198]: Sounds good! I'm looking forward to seeing it 👍

**Pull request merged**